### PR TITLE
[Do NOT Review] [Az.Network] Added a new endpoint type 'AzureArcNetwork' in Connection Monnitor

### PR DESCRIPTION
**Pull request by v-sapanchal**
## Description

Added a new endpoint type 'AzureArcNetwork' in command 'New-AzNetworkWatcherConnectionMonitorEndpointObject' along with some mandatory parameters for this type and consume it in 'New-AzNetworkWatcherConnectionMonitor' .

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Network/Network/Network.csproj [hunk 1]
```diff
@@ -41,10 +42,6 @@
     <Folder Include="RouteServerPeer\" />
   </ItemGroup>
   
-  <ItemGroup>
-    <ProjectReference Include="..\Network.Management.Sdk\Network.Management.Sdk.csproj" />
-  </ItemGroup>
-
```
Comment (YanaXu) [inline src/Network/Network/Network.csproj:47]: Please revert the change to remove the local built "Network.Management.Sdk". And remove the nupkg files under tools/LocalFeed.

**Comment by YanaXu** [comment]
Hi @v-sapanchal, please update the PR according to my comment and fix CI.

File: src/Network/Network/Network.csproj (hunk 1 above)
Comment (v-sapanchal) [reply src/Network/Network/Network.csproj:47]: Hi @YanaXu ,we have kind of dependency on updated SDK related PR ([https://github.com/Azure/azure-rest-api-specs/pull/24965](https://github.com/Azure/azure-rest-api-specs/pull/24965/files)),
Once that PR get merge then we will update it here.
If you have any suggestion to resolve this dependency, please suggest us.
Thank you.

File: src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorEndPointObjectCommand.cs [hunk 2]
```diff
@@ -182,6 +199,20 @@ public class NewAzureNetworkWatcherConnectionMonitorEndpointObjectCommand : Conn
         [PSArgumentCompleter("Default", "Low", "BelowAverage", "Average", "AboveAverage", "Full")]
         public string CoverageLevel { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Azure Arc Network.",
+             ParameterSetName = "AzureArcNetwork")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
```
Comment (kumaam) [inline src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorEndPointObjectCommand.cs:207]: Should we use LocationDetails or Location only ?

Help message require an update

**Comment by YanaXu** [comment]
Hi @v-sapanchal, I update the title to add the tag "Do NOT Review". Please remove it when your PR is ready.

**Comment by YanaXu** [comment]
Hi @v-sapanchal , I've converted this PR to a draft one. Please click "Ready for review" on this page when it's ready.
